### ISSUE TITLE
🚀 Feature: Filter `Location`s by targets in the main catalog table

### ISSUE THREAD

**Issue by dotboris** [backstage/backstage#23098]
### 🔖 Feature description

The main catalog table (ex: https://demo.backstage.io/catalog?filters%5Bkind%5D=location&filters%5Buser%5D=all) has a filter on the top-right. Currently, this filter seems to only search the location name and not the location target.

Here's an example to illustrate:

When I look for `dice-roller` in the backstage demo:
![image](https://github.com/backstage/backstage/assets/1874242/19f5b0a7-c6f1-4bbf-b62c-9e289f29b4b4)

I should be able to put `dice-roller` in the filter box and find it easily. Unfortunately, this doesn't work:
![image](https://github.com/backstage/backstage/assets/1874242/eada6c7a-248a-4bc8-b0e3-eb9941d56e45)

You should be able to validate this for yourself using the [backstage demo](https://demo.backstage.io/catalog?filters%5Bkind%5D=location&filters%5Buser%5D=all)

### 🎤 Context

This is a must when loading the catalog from GitHub using the `GithubEntityProvider`. The location names it generates are all `generated-{hash}`.

In my case, I sometimes have to debug things in the catalog and getting a hold the right `Location` is very useful to figure out what's going on or to do manual manipulations like deleting the location.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by dotboris** [comment]
I did some testing and I found a solution:

```patch
diff --git a/plugins/catalog/src/components/CatalogTable/CatalogTable.tsx b/plugins/catalog/src/components/CatalogTable/CatalogTable.tsx
index 006210bb1d6a..cd48bb1c3e64 100644
--- a/plugins/catalog/src/components/CatalogTable/CatalogTable.tsx
+++ b/plugins/catalog/src/components/CatalogTable/CatalogTable.tsx
@@ -109,6 +109,7 @@ const defaultColumnsFunc: CatalogTableColumnsFunc = ({ filters, entities }) => {
       case 'location':
         return [
           columnFactories.createSpecTypeColumn({ hidden: !showTypeColumn }),
+          {field: 'entity.spec.target', hidden: true, searchable: true},
           columnFactories.createSpecTargetsColumn(),
         ];
       default:
```

This issue happens because of the discrepancy between `spec.target` and `spec.targets`. The column defined by `columnFactories.createSpecTargetsColumn()` specified `field: 'entity.spec.targets'`. When the table handles the searching, it ends up looking in the `spec.targets` field but not in the `spec.target` field. In my example above, that location happens to define `spec.target` which is not getting searched through since it's not in the columns.

The fix here is to add a hidden & searchable column for `spec.target`. That way the search can look through it as well.

**Comment by dotboris** [comment]
I just tested this now and the issue is now fixed. I believe that it was released in 1.24
